### PR TITLE
Asm import fix

### PR DESCRIPTION
**Pull request by gerazo**
https://github.com/dkrupp/clang/issues/70

### CODE REVIEW

File: lib/AST/ASTImporter.cpp [hunk 1]
```diff
@@ -5244,40 +5244,35 @@ Stmt *ASTNodeImporter::VisitGCCAsmStmt(GCCAsmStmt *S) {
   SmallVector<IdentifierInfo *, 4> Names;
   for (unsigned I = 0, E = S->getNumOutputs(); I != E; I++) {
     IdentifierInfo *ToII = Importer.Import(S->getOutputIdentifier(I));
-    if (!ToII)
-      return nullptr;
```
Comment (dkrupp) [inline lib/AST/ASTImporter.cpp]: add an assert here

**Comment by dkrupp** [comment]
check ast reader/writer and check how asm works in non-xtu case

**Comment by gerazo** [comment]
Here is the refined solution. ASTReader and ASTWriter works fine. These kind of asms are never parsed correctly. The parser intentionally puts nullptr into the identifier ID https://github.com/dkrupp/clang/blob/ctu-master/lib/Parse/ParseStmtAsm.cpp#L843 and serilaizers handle this correctly. Even without XTU, the parser never reads non-null information out of the source code. So this behaviour is intended. I've put back the breaking of stmt to trigger assert when it is applicable, but the identifier ids can be sometimes null, so even asserting on it would be a mistake as the original parser also doesn't do it.
I still have to test the solution on big projects.

**Comment by gerazo** [comment]
On openssl, the bug is completely solved.

**Pull request closed without merging**